From a36b0bfad3dd26fb289ada292db8d8c48dbce22c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sun, 21 Apr 2019 22:25:03 +0200
Subject: [PATCH 12/29] sd-bus: when running user find function don't trust the
 value to be initialized

The find function is externally provided, and we shouldn't trust that the
authors remember to set the output parameter in all cases.

(cherry picked from commit bf135d288d11c1da6b053dfe26ffa708031d544f)
---
 src/libsystemd/sd-bus/bus-objects.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsystemd/sd-bus/bus-objects.c b/src/libsystemd/sd-bus/bus-objects.c
index ce2cb94bde..c676b23ebb 100644
--- a/src/libsystemd/sd-bus/bus-objects.c
+++ b/src/libsystemd/sd-bus/bus-objects.c
@@ -22,7 +22,7 @@ static int node_vtable_get_userdata(
                 sd_bus_error *error) {
 
         sd_bus_slot *s;
-        void *u, *found_u;
+        void *u, *found_u = NULL;
         int r;
 
         assert(bus);
-- 
2.21.0

